modelskill/matching.py

```diff
@@ -334,26 +354,44 @@ def match_time(
     """
     obs_name = "Observation"
     mod_names = list(raw_mod_data.keys())
-    idxs = [m.index for m in raw_mod_data.values()]
+    idxs = [m.time for m in raw_mod_data.values()]
     period = _get_global_start_end(idxs)
 
     assert isinstance(observation, (PointObservation, TrackObservation))
     gtype = "point" if isinstance(observation, PointObservation) else "track"
     observation = observation.copy()
     observation.trim(period.start, period.end)
 
-    data = observation.data.copy()
+    data = observation.data  # .copy() #already a copy?
     data.attrs["name"] = observation.name
     data = data.rename({observation.name: obs_name})
 
-    for name, mdata in raw_mod_data.items():
-        df = _model2obs_interp(observation, mdata, max_model_gap)
-        if gtype == "track":
-            # TODO why is it necessary to do mask here? Isn't it an error if the model data is outside the observation track?
-            df_obs = observation.data.to_pandas()  # TODO: xr.Dataset
-            _mask_model_outside_observation_track(name, df, df_obs)
+    for name, mr in raw_mod_data.items():
+        # df = _model2obs_interp(observation, mdata, max_model_gap)
+        if isinstance(observation, PointObservation):
+            assert len(observation.time) > 0
+            mri = mr.interp_time(new_time=observation.time)
+        else:
+            # It doesn't make sense to interpolate track data in time
```

Comment:
@ecomodeller could you please help with this part? We still need matching for track data, just a different kind than for point data. For track data we should make sure t, x and y match (within a tolerance) - the model and observation series will not necessarily have same length or exact same data even though the observation data was used to extract data from the dfsu model result. The obs data will typically be covering more than the model domain. Also, we have in the past had problems around rounding of position and time data.

Reply:
I think we should re-evaluate how much of this track matching logic that belongs in the core of `modelskill`.

If the observation data was used to extract model data, they should already match, otherwise the problem is in the extraction process. I would assume that it would be possible to get a matched dataset in connection with the extraction and then supply this pre-matched data to `modelskill` with the `modelskill.from_matched(...)` function.

Reply:
I made an attempt of creating some matching logic, but it is difficult since I don't fully understand the requirements.

I can understand two ends of the spectrum. 
1. Extracted data that already matches, i.e. equal number of rows
2. Intersection of observation and model data in 3d (t,y,x) with a buffer applied to each point

In order to something in between, I need more information on what is expected from the data.